packages/ui/src/multiplying-architecture/KaotoEditorChannelApi.ts

```diff
@@ -12,4 +12,32 @@ export interface KaotoEditorChannelApi extends KogitoEditorChannelApi {
    * Returns the Kaoto VSCode settings defined.
    */
   getVSCodeKaotoSettings(): Promise<ISettingsModel>;
+
+  /**
+   * Get metadata querying a Kaoto metadata file.
+   * @param path The path of the file being edited
+   * @param key The key to retrieve the preferences
+   */
+  getMetadata<T>(path: string, key: string): Promise<T>;
+
+  /**
+   * Save preferences to a Kaoto metadata file.
+   * @param path The path of the file being edited
+   * @param key The key to set the preferences
+   * @param metadata The preferences to be saved
+   */
+  setMetadata<T>(preferencesFilePath: string, key: string, metadata: T): Promise<void>;
+
+  /**
+   * Retrieve resource content
+   * @param path The path of the resource, relative to the file being edited
```

Comment:
just a note to ensure that you have in mind that this info will need to be computed on your side.
In maven based project, I think the path which is mentioned in the camel route is the one from the classpath, so from src/main/resources usually.

Reply:
Since we would need to use a `File` picker of some sorts, the expectation would be to intercept that filepath and make it relative to the open file

Reply:
This could work only on first edit.
When reopening the editor, it won't go through the File picker.

Reply:
no, it won't, but the path will be already stored if needed

Reply:
when moving from standalone to a Maven projects, need to check where the xslt files are moved and if the saved path needs to be updated then

Reply:
I think this conversation will need to happen once we get more information about the topic, at the moment of the implementation. I'll remove the comment `relative to the file being edited` so we can move forward.